### ISSUE TITLE
[API] Client registration doesn't work anymore

### ISSUE THREAD

**Issue by cmrd-senya** [diaspora/diaspora#7059]
I'm not sure if this issue is valid.

#6182 contains client registration call:
```
curl http://localhost:3000/api/openid_connect/clients -d "redirect_uris[]=http://localhost:3000/&client_name=<your_client_name>"
```

After the recent commit 6ad9000f8c16f6758c39b59412a4b6978910d690 it stopped working. It returns the following instead of expected json.
```
<html><body>You are being <a href="http://development.diaspora.local/users/sign_in">redirected</a>.</body></html>
```

I guess it would work if we log in as some user, but do we need a logged user to register a client on pod?

**Comment by KentShikama** [comment]
Originally, I planned to allow anyone but it might actually be a good idea to only allow logged in users: it would essentially be a substitute of the access token describe below

https://openid.net/specs/openid-connect-registration-1_0.html#ClientRegistration

**Comment by KentShikama** [comment]
As a side note, I've added "Add rate-limiting for dynamic client registration" to https://github.com/diaspora/diaspora/issues/6182

**Comment by cmrd-senya** [comment]
I just tested it with a pre-logged user, it still fails due to CSRF error. Diaspora log:
```
[2016-09-04T22:35:40] INFO  PID-3759 TID-69969072893460 Rails: Started POST "/api/openid_connect/clients" for 127.0.0.1 at 2016-09-04 22:35:40 +0000
[2016-09-04T22:35:40] INFO  PID-3759 TID-69969072893460 ActionController::Base: Processing by Api::OpenidConnect::ClientsController#create as */*
[2016-09-04T22:35:40] INFO  PID-3759 TID-69969072893460 ActionController::Base:   Parameters: {"client_name"=>"a web application via omniauth-openid-connect", "redirect_uris"=>["http://localhost:3000/users/auth/diaspora/callback"], "client"=>{"client_name"=>"a web application via omniauth-openid-connect", "redirect_uris"=>["http://localhost:3000/users/auth/diaspora/callback"]}}
[2016-09-04T22:35:40] WARN  PID-3759 TID-69969072893460 Api::OpenidConnect::ClientsController: Can't verify CSRF token authenticity
[2016-09-04T22:35:40] INFO  PID-3759 TID-69969072893460 ActionController::Base: Redirected to http://development.diaspora.local/users/sign_in
[2016-09-04T22:35:40] INFO  PID-3759 TID-69969072893460 ActionController::Base: Completed 302 Found in 4ms (ActiveRecord: 0.0ms)
```

Actually I think this feature can't  rely on the fact that user is authenticated. To get this information from pod, we must make an API request. And to make an API request we must be registered. It must work without anyone logged. Therefore it is a bug.

**Comment by cmrd-senya** [comment]
This seem to be fixed with
```diff
diff --git a/app/controllers/api/openid_connect/clients_controller.rb b/app/controllers/api/openid_connect/clients_controller.rb
index 0a6f7ba..4c22f44 100644
--- a/app/controllers/api/openid_connect/clients_controller.rb
+++ b/app/controllers/api/openid_connect/clients_controller.rb
@@ -1,6 +1,8 @@
 module Api
   module OpenidConnect
     class ClientsController < ApplicationController
+      protect_from_forgery
+
       rescue_from OpenIDConnect::HttpError do |e|
         http_error_page_as_json(e)
       end
```

Hopefully, gonna prepare a PR tomorrow.

**Comment by cmrd-senya** [comment]
This was fixed by #7062, closing.
